src/languages/typescript.js

```diff
@@ -222,8 +223,14 @@ export default function(hljs) {
       {
         begin: '\\.' + hljs.IDENT_RE, relevance: 0 // hack: prevents detection of keywords after dots
       },
+      {
+        begin: '\\b' + IDENT_RE + ':', relevance: 0 // prevents detection of keywords in property name
```

Comment:
The better way to do this might be the very complex way we do it in JS now, and also to highlight these as `attr`.  That's probably something we should move into a `js-like` library so that both JS and TS should share it.

I think the best thing at this point might be to scale this back to JUST adding the `readonly` keyword and to open an issue (with examples, please) for the other issues so they can best be discussed how to solve them keeping the bigger picture in mind (like JS grammar, etc)